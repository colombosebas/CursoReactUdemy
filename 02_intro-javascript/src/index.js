const activo = true;
let mensaje = '';
if (activo){
    mensaje = 'Activo';
}else{
    mensaje = 'Inactivo';
}
const mensaje2 = (activo) ? 'Activo' : 'Inactivo';
const mensaje3 = activo && 'Activo';
