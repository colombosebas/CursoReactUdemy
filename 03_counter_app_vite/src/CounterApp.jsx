import PropTypes from 'prop-types';

const handleAdd = (event) => console.log(event)

export const CounterApp = ({value}) => {
    return(
        <>
            <h1>CounterApp</h1>
            <h2>{value}</h2>
            <button onClick={handleAdd}>1</button>
        </>
    );
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired,
}