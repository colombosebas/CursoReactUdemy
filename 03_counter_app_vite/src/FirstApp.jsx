 import PropTypes from 'prop-types';

const newMessage = 'Sebastián Colombo'
const newObject = {
    message: 'Hola mundo',
    subject: 'Terricola'
};

 const funcion = () => 'Mi función';

export const AppFirst = ({title,subTitle}) => {
    return(
        <>
            <h1 data-testid='test-title'>{title}</h1>
            <h2>Sabías {newMessage}</h2>
            <h3>{subTitle}</h3>
        </>
    );
}

AppFirst.propTypes ={
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

AppFirst.defaultProps = {
    title: 'No hay título',
    subTitle: 123
}

//Esto da error, no se pueden imprimir objetos
// export const AppFirst2 = () => {
//     return(<h2>{newObject}</h2>);
// }

// Así si funciona la impresión de objetos
export const AppFirst2 = () => {
    return(<h2>{JSON.stringify(newObject)}</h2>);
}


// //Acá llamo a una función, ojo que no puede ser una función asíncrona
export const AppFirst3 = () => {
    return(<h3>{funcion()}</h3>);
}