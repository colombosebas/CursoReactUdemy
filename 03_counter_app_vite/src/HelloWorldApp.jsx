export function App() {
    //documento.createElement.....
    return (<h1>Hola Mundo!!!!</h1>);
}

//Otra forma de hacerlo
/*
export const App = () => {
    return(<h1>Hola Mundo!!!!</h1>);
}
*/

//Otra forma de hacerlo
/*
export const App = () => (<h1>Hola Mundo!!!!</h1>);
*/

export const App2 = () => {
    return(<h2>Soy Sebastián Colombo</h2>);
}

//si quiero poner dos cosas lo puedo hacer así (no es la mejor forma):
export const App3 = () => {
    return(
    <div>
        <h3>App3</h3>
        <p>Primera forma</p>
    </div>
    );
}

//Esta forma es mejor
import {Fragment} from 'react'
export const App4 = () => {
    return(
    <Fragment>
        <h3>App4</h3>
        <p>Segunda Forma</p>
    </Fragment>
    );
}

//Pero esta forma es la mejormejor
export const App5 = () => {
    return(
    <>
        <h3>App5</h3>
        <p>Tercera Forma</p>
    </>
    );
}