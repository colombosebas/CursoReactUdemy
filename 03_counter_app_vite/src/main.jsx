import React from 'react';
import ReactDOM from 'react-dom/client';
import {AppFirst, AppFirst2, AppFirst3} from './FirstApp';
import {CounterApp} from './CounterApp';
import './style.css';

 ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
         <AppFirst title="Soy Goku"/>
         <AppFirst2/>
         <AppFirst3/>
         <CounterApp value={51}/>
     </React.StrictMode>
 )