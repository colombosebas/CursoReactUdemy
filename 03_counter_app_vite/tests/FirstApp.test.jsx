import { render } from "@testing-library/react";
import { AppFirst } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {
  // test('debe hacer match con el snapshot', () => {

  //       const title = "Soy Goku";
  //       const {container} = render(<AppFirst title={title}/>)

  //       expect(container).toMatchSnapshot();
  // });

  test('Debe mostrar el tÃ­tulo en un h1', () => {
    
    const title = "Soy Goku";
    const {container, getByText, getByTestId} = render(<AppFirst title={title}/>)
    expect(getByText(title)).toBeTruthy();

    // const h1 = container.querySelector('h1');
    // expect(h1.innerHTML).toBe(title);

    expect(getByTestId('test-title').innerHTML).toBe(title)
  })
   

});
