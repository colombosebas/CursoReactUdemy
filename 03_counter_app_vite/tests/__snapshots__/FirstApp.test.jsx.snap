// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Soy Goku
  </h1>
  <h2>
    Sabías 
    Sebastián Colombo
  </h2>
  <h3>
    123
  </h3>
</div>
`;
